SeeSharp/Geometry/PlyMesh.cs

```diff
@@ -0,0 +1,454 @@
+                        string name = parts[4];
+
+                        if (!IsAllowedVertCountType(countType)) {
+                            errors.Add($"Only 'property list uchar int' is supported. Ignoring {countType}");
+                            continue;
+                        }
+
+                        if (!IsAllowedVertIndType(indType)) {
+                            errors.Add($"Only 'property list uchar int' is supported. Ignoring {indType}");
+                            continue;
+                        }
+
+                        if (name == "vertex_indices")
+                            header.IndElem = facePropCounter - 1;
+                    } else {
+                        errors.Add($"Only float or list properties allowed. Ignoring {type}");
+                        ++header.VertexPropCount;
+                    }
+                } else if (action == "end_header") {
+                    break;
+                } else {
+                    errors.Add($"Unknown header entry {action}");
+                }
+            }
+
+            return errors;
+        }
+
+        /// <summary>
+        /// Will parse the whole file, starting with the header and following up with the data region
+        /// </summary>
+        /// <param name="stream"></param>
+        /// <returns>List of errors</returns>
+        private List<string> ParsePlyFile(MixReader stream) {
+            List<string> errors = new();
+
+            PlyHeader header = new();
+            errors.AddRange(ParsePlyHeader(stream, ref header));
+
+            if (!header.HasVertices || !header.HasIndices) {
+                errors.Add("Reached end of stream without data");
+                return errors;
+            }
+
+            // Setup reader
+            IDataReader reader;
+            if (header.IsBinary) {
+                reader = new BinaryDataReader(header, stream);
+            } else {
+                reader = new AsciiDataReader(header, stream);
+            }
+
+            // Reserve memory
+            Vertices.Capacity = header.VertexCount;
+            if (header.HasNormals) Normals.Capacity = header.VertexCount;
+            if (header.HasUVs) Texcoords.Capacity = header.VertexCount;
+
+            // Read per vertex stuff
+            for (int i = 0; i < header.VertexCount; ++i) {
+                (Vector3 vertex, Vector3 normal, Vector2 tex) = reader.ReadPerVertexLine();
+
+                Vertices.Add(vertex);
+                if (header.HasNormals) Normals.Add(normal);
+                if (header.HasUVs) Texcoords.Add(tex);
+            }
+
+            // Read per face indices
+            if (header.IndElem != 0) {
+                errors.Add("No support for multiple face properties. Assuming first entry to be the list of indices");
+            }
+
+            for (int i = 0; i < header.FaceCount; ++i) {
+                Faces.Add(reader.ReadFaceLine());
+                // TODO: Maybe enforce triangle (or quads) only?
```

Comment:
Aren't you doing that by triangulating? Or what do you mean here?

Reply:
Nahh, just forgot to remove the comment 🤣